library/ssl_tls13_client.c

```diff
@@ -1065,8 +1250,13 @@ static int ssl_tls13_parse_server_hello( mbedtls_ssl_context *ssl,
 #if defined(MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED)
             case MBEDTLS_TLS_EXT_KEY_SHARE:
                 MBEDTLS_SSL_DEBUG_MSG( 3, ( "found key_shares extension" ) );
-                if( ( ret = ssl_tls13_parse_key_share_ext( ssl,
-                                            p, p + extension_data_len ) ) != 0 )
+                if( is_hrr )
```

Comment:
Pre-existing but now we send the key share extension only if `mbedtls_ssl_conf_tls13_some_ephemeral_enabled()` is true (see write_client_hello()) thus here if we receive this extension and  `mbedtls_ssl_conf_tls13_some_ephemeral_enabled()` is false we should abort the handshake and send an "unsupported_extension" alert. We should try to factorize all the "unsupported_extension" cases: add an "unsupported_extension_found" flag and go to an exit label at the end of the function?